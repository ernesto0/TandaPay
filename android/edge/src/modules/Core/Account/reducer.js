// @flow

import type { EdgeAccount } from 'edge-core-js'
import { type Reducer } from 'redux'

import type { Action } from '../../ReduxTypes.js'

export type AccountState = EdgeAccount | Object

export const initialState: AccountState = {}

const accountReducer = (state = initialState, action: Action) => {
  switch (action.type) {
    case 'ACCOUNT/LOGGED_IN': {
      if (!action.data) throw new Error('Invalid Action')
      return action.data.account
    }

    default:
      return state
  }
}

export const account: Reducer<AccountState, Action> = (state, action: Action) => {
  if (action.type === 'LOGOUT' || action.type === 'DEEP_LINK_RECEIVED') {
    state = undefined
  }
  return accountReducer(state, action)
}
