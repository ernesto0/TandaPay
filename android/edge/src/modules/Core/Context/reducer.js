// @flow

import { type DiskletFolder } from 'disklet'
import type { EdgeContext } from 'edge-core-js'
import { type Reducer } from 'redux'

import { type Action, type Username } from '../../ReduxTypes.js'

export type ContextState = {
  context: EdgeContext | Object,
  folder: DiskletFolder | Object,
  usernames: Array<Username>,
  nextUsername: Username
}

const initialState = {
  context: {},
  folder: {},
  nextUsername: '',
  usernames: []
}

export const context: Reducer<ContextState, Action> = (state = initialState, action: Action) => {
  switch (action.type) {
    case 'CORE/CONTEXT/ADD_CONTEXT': {
      if (!action.data) throw new Error('Invalid action')
      const context: EdgeContext = action.data.context
      const folder: DiskletFolder = action.data.folder
      return {
        ...state,
        context,
        folder
      }
    }

    case 'CORE/CONTEXT/ADD_USERNAMES': {
      if (!action.data) throw new Error('Invalid action')
      const { usernames } = action.data
      return {
        ...state,
        usernames
      }
    }

    case 'CORE/CONTEXT/DELETE_LOCAL_ACCOUNT_REQUEST': {
      if (!action.data) throw new Error('Invalid action')
      const { username } = action.data
      return {
        ...state,
        usernames: state.usernames.filter(name => name !== username)
      }
    }

    case 'DEEP_LINK_RECEIVED':
    case 'LOGOUT': {
      if (!action.data) throw new TypeError('Invalid action')
      const { username } = action.data
      return {
        ...state,
        nextUsername: username || ''
      }
    }

    default:
      // $FlowFixMe
      return state
  }
}
