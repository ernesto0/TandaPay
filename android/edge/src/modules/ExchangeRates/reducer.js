// @flow

import { type Reducer } from 'redux'

import type { Action } from '../ReduxTypes.js'

export type ExchangeRatesState = {
  [string]: number
}

const initialState = {}

export const exchangeRates: Reducer<ExchangeRatesState, Action> = (state = initialState, action: Action) => {
  switch (action.type) {
    case 'EXCHANGE_RATES/UPDATE_EXCHANGE_RATES':
      if (!action.data) throw new Error('Invalid action')
      return action.data.exchangeRates
    case 'LOGOUT':
      return initialState
    case 'DEEP_LINK_RECEIVED':
      return initialState
    default:
      return state
  }
}
