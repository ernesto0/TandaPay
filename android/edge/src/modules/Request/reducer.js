// @flow

import type { Action } from '../ReduxTypes.js'

export type RequestState = {
  receiveAddress: {
    publicAddress: string,
    amountSatoshi: number,
    metadata: {
      payeeName: string,
      category: string,
      notes: string,
      amountFiat: number,
      bizId: null,
      miscJson: string
    }
  }
}

const initialState: RequestState = {
  receiveAddress: {
    publicAddress: '',
    amountSatoshi: 0,
    metadata: {
      payeeName: '',
      category: '',
      notes: '',
      amountFiat: 0,
      bizId: null,
      miscJson: ''
    }
  }
}

export const request = (state: RequestState = initialState, action: Action): RequestState => {
  let receiveAddress
  let amountSatoshi
  let amountFiat

  switch (action.type) {
    case 'UPDATE_RECEIVE_ADDRESS_SUCCESS': {
      if (!action.data) throw new Error('Invalid action')
      receiveAddress = action.data.receiveAddress
      return {
        ...state,
        receiveAddress
      }
    }

    case 'UPDATE_AMOUNT_REQUESTED_IN_CRYPTO': {
      if (!action.data) throw new Error('Invalid action')
      amountSatoshi = action.data.amountSatoshi
      return {
        ...state,
        receiveAddress: {
          ...state.receiveAddress,
          amountSatoshi
        }
      }
    }

    case 'UPDATE_AMOUNT_REQUESTED_IN_FIAT': {
      if (!action.data) throw new Error('Invalid action')
      amountFiat = action.data.amountFiat
      return {
        ...state,
        receiveAddress: {
          ...state.receiveAddress,
          metadata: {
            ...state.receiveAddress.metadata,
            amountFiat
          }
        }
      }
    }

    default:
      return state
  }
}
