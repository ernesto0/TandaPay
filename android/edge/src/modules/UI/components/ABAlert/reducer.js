// @flow

import { type Reducer, combineReducers } from 'redux'

import { type Action } from '../../../ReduxTypes.js'

export type ABAlertState = {
  view: boolean,
  syntax: {
    title: string,
    message: string,
    buttons: Array<{ title: string, message: string }>
  }
}

const initialSyntax = {
  title: '',
  message: '',
  buttons: []
}

const view = (state = false, action: Action): boolean => {
  switch (action.type) {
    case 'OPEN_AB_ALERT': {
      return true
    }

    case 'CLOSE_AB_ALERT': {
      return false
    }

    default:
      return state
  }
}

const syntax = (state = initialSyntax, action: Action): $PropertyType<ABAlertState, 'syntax'> => {
  switch (action.type) {
    case 'OPEN_AB_ALERT': {
      if (action.data == null) throw new TypeError('Invalid action')
      return action.data
    }

    case 'CLOSE_AB_ALERT': {
      return initialSyntax
    }

    default:
      return state
  }
}

export const ABAlert: Reducer<ABAlertState, Action> = combineReducers({
  syntax,
  view
})
