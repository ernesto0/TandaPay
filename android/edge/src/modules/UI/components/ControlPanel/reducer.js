// @flow

import { type Reducer, combineReducers } from 'redux'

import { type Action, type Username } from '../../../ReduxTypes.js'

export type ControlPanelState = {
  +usersView: boolean,
  +selectedUser: Username | null
}

const usersView = (state = false, action: Action): boolean => {
  switch (action.type) {
    case 'OPEN_SELECT_USER': {
      return true
    }

    case 'CLOSE_SELECT_USER': {
      return false
    }

    default:
      return state
  }
}

const selectedUser = (state = null, action: Action): Username | null => {
  switch (action.type) {
    case 'LIST_USER_USER_SIDE_MENU': {
      if (!action.data) throw new Error('Invalid action')
      return action.data[0]
    }

    default:
      return state
  }
}

export const controlPanel: Reducer<ControlPanelState, Action> = combineReducers({
  usersView,
  selectedUser
})
