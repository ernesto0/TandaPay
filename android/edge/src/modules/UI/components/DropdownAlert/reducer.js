// @flow

import type { Action } from '../../../ReduxTypes.js'

const initialState = {
  visible: false,
  type: '',
  title: '',
  message: ''
}

type DropdownAlertState = {
  visible: boolean,
  type: string,
  title: string,
  message: string
}

export const dropdownAlert = (state: DropdownAlertState = initialState, action: Action) => {
  switch (action.type) {
    case 'UI/COMPONENTS/DROPDOWN_ALERT/DISPLAY_DROPDOWN_ALERT': {
      if (!action.data) throw new Error('Invalid action')
      const { type, title, message } = action.data

      return {
        visible: true,
        type,
        title,
        message
      }
    }

    case 'UI/COMPONENTS/DROPDOWN_ALERT/DISMISS_DROPDOWN_ALERT': {
      return initialState
    }

    default:
      return state
  }
}

export default dropdownAlert
