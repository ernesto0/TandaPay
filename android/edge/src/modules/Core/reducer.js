// @flow

import { type Reducer, combineReducers } from 'redux'

import { type DeepLinkingState, deepLinking } from '../../reducers/core/DeepLinkingReducer.js'
import { type EdgeLoginState, edgeLogin } from '../../reducers/core/EdgeLoginReducer.js'
import { type Action } from '../ReduxTypes.js'
import { type AccountState, account } from './Account/reducer.js'
import { type ContextState, context } from './Context/reducer.js'
import { type WalletsState, wallets } from './Wallets/reducer.js'

export type CoreState = {
  +account: AccountState,
  +context: ContextState,
  +wallets: WalletsState,
  +edgeLogin: EdgeLoginState,
  +deepLinking: DeepLinkingState
}

export const core: Reducer<CoreState, Action> = combineReducers({
  account,
  context,
  deepLinking,
  edgeLogin,
  wallets
})
