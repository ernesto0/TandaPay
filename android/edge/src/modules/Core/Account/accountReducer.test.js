/* eslint-disable flowtype/require-valid-file-annotation */

/* globals describe test expect */

import { account as accountReducer, initialState } from './reducer.js'

describe('account', () => {
  test('initialState', () => {
    const expected = initialState
    const actual = accountReducer(undefined, {})

    expect(actual).toEqual(expected)
  })

  test('LOGGED_IN', () => {
    const account = { id: '123123', activeWalletIds: ['1', '2', '3'] }
    const action = { type: 'ACCOUNT/LOGGED_IN', data: { account } }
    const actual = accountReducer(initialState, action)

    expect(actual).toMatchSnapshot()
  })
})
