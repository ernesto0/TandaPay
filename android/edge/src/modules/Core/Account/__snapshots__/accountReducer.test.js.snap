// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`account LOGGED_IN 1`] = `
Object {
  "activeWalletIds": Array [
    "1",
    "2",
    "3",
  ],
  "id": "123123",
}
`;
